### ISSUE TITLE
error: CUDA library was not found 

### ISSUE THREAD

**Issue by mgoundge11** [horovod/horovod#734]
Hi , I am getting this error while installing horovod ,
Please specify correct CUDA location with the HOROVOD_CUDA_HOME environment variable or combination of HOROVOD_CUDA_INCLUDE and HOROVOD_CUDA_LIB environment variables.
    
    HOROVOD_CUDA_HOME - path where CUDA include and lib directories can be found
    HOROVOD_CUDA_INCLUDE - path to CUDA include directory
    HOROVOD_CUDA_LIB - path to CUDA lib directory
I have specified the same in my bashrc file but still it is unable to find the CUDA libraries. Please help, I'm stuck here. thanks.

**Comment by alsrgv** [comment]
@mgoundge11, can you try installing Horovod like this:
```
$ [other flags] HOROVOD_CUDA_HOME=/path/to/cuda pip install --no-cache-dir horovod
```

**Comment by mgoundge11** [comment]
Thanks @alsrgv for reply. I solved it by using following command,
sudo apt-get install nvidia-cuda-toolkit , I finished installing horovod in two ubuntu systems.

But I'm unable to run the keras_mnist example provided here in the link,
https://github.com/uber/horovod/blob/master/examples/keras_mnist.py
I wanted to run it on two ubuntu servers with single gpu each.
I given parameters as below,
$ mpirun -np 2 \
    -H server1:1,server2:1, \
    -bind-to none -map-by slot \
    -x NCCL_DEBUG=INFO -x LD_LIBRARY_PATH -x PATH \
    -mca pml ob1 -mca btl ^openib \
    python keras_mnist.py
do I need to give ip addresses of server1 and server2? I am using tcp communication.

**Comment by alsrgv** [comment]
@mgoundge11, yes, you do, and you need to make sure that you can ssh into both servers w/o password.

**Comment by mgoundge11** [comment]
Thanks @alsrgv

**Comment by mgoundge11** [comment]
Hi @alsrgv I installed horovod on two ubuntu servers with single GPU each but after I execute  the command to run horovod it is stuck without any error.
I made the ssh password less for both the servers.
Pls help on this I am stuck here.

**Comment by mgoundge11** [comment]
Hi @alsrgv Actually I'm able set up the two machine with password less ssh, mpi 4.0 configuration.
I run the following command 
 mpirun --prefix /home/ges/openmpi -np 2 -H server1:1,server2:1 -bind-to none -map-by slot -x NCCL_DEBUG=INFO -x LD_LIBRARY_PATH -x PATH -mca pml ob1 -mca btl_tcp_if_include enp0s31f6 -x NCCL_SOCKET_IFNAME=enp0s31f6 python3 horovod_keras_mnist.py

I changed only Adadelta optimizer to Adam in keras_mnist.py example 
But I'm facing the following error,

Caused by op 'training/Adam/DistributedAdam_Allreduce/HorovodAllreduce_training_Adam_gradients_dense_2_MatMul_grad_MatMul_1_0', defined at:
  File "horovod_keras_mnist.py", line 92, in <module>
    validation_data=(x_test, y_test))
nknownError (see above for traceback): ncclCommInitRank failed: unhandled system error
	 [[Node: training/Adam/DistributedAdam_Allreduce/HorovodAllreduce_training_Adam_gradients_dense_2_MatMul_grad_MatMul_1_0 = HorovodAllreduce[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"](training/Adam/gradients/dense_2/MatMul_grad/MatMul_1)]]

**Comment by alsrgv** [comment]
@mgoundge11, great!  Could you make [a gist](https://gist.github.com) with the full script output?

**Comment by mgoundge11** [comment]
hi @alsrgv thanks. I am able to run the keras_mnist example on the two systems with single gpu each.   
I kept same data and source code in both systems but I'm unable understand how it's using data for training from each system and its taking more time than single machine training?
how to data parallelism works in horovod can u pls explain ?

**Comment by alsrgv** [comment]
@mgoundge11, MNIST example has very small iteration time which does not scale well beyond a single server.  If you switch to training bigger models, you should see better scaling efficiency.
